.github/workflows/verify_samples.yml

```diff
@@ -0,0 +1,50 @@
+name: Build Samples
+
+on: [push, pull_request]
+
+jobs:
```

Comment:
nit: I prefer jobs / OS lists in this order:

- Windows
- macOS
- Linux (Ubuntu in this case)

For user-facing stuff, Windows and macOS are more popular than Linux. For CI stuff, we should still keep it for consistency, and it also reflects what's most important for most developers (as the order of OS popularity is the other way around).

Also Windows and macOS are more likely to have artifacts, so keeping those OS on top would be nice.

Reply:
Note: XQEMU actually split platforms into different workflows (my preferred order is lost then); different workflows is useful for having different CI badges (not sure if this is possible otherwise).
I'm not sure, but tasks might also run in parallel then (not sure how jobs handle this)?

Reply:
Even if unintentional it has been the preferred order on XQEMU to run Linux (Ubuntu) then macOS on travis-ci, i'd like to keep with that and add Windows at the bottom.

Reply:
As I have said in my previous comment: XQEMU splits platforms into workflows with the Actions CI. The workflows seems to be alphabetical order (or round-robin or something?) - I don't think we order it explicitly anymore. So there's no "preferred order". The implicit order is fine - there's no point in adding more code just to have the explicit order.

However, generally, if we explicitly list platforms I ensure we do this in this order: Windows, macOS, Linux (which would be the preferred order).

*(We did have some older lists in XboxDev and XQEMU which might still be unordered, but most user facing changes had been changed in the past, and new lists generally follow this order as I bring it up in review)*

---

Splitting into different workflows for this nxdk CI might also be beneficial for different CI badges (also explained in my previous comment) - it will be implicit order then; but we might have other benefits (different badges per platform + possibly faster jobs?).

Reply:
I disagree in this ordering.

Reply:
The ordering was changed.
This discussion remains unresolved because splitting workflows per platform might have benefits (but wasn't done yet):

- Probably easier to add more steps for things other than building samples.
- [GitHub badges are per workflow](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository) (ability to separate platforms). At the time of writing, there don't seem to be per-job badges.
- Separate the configuration more cleanly into separate files (which might avoid git merge conflicts).

That said: I could tolerate the current PR (in fact, I prefer how the GitHub Actions CI displays this workflow over the one proposed for XQEMU which is per-platform workflows).

Also, I'm not confident in either solution; I simply lack the understanding of how GitHub intends users to use GitHub Actions. I don't fully understand the differences between workflows, jobs and the actions themselves.

However, I worry we'd change this to per-platform-workflow soon. My understanding is, that @mborgerson had also expressed that per-platform workflows have benefits (or did I misunderstand this?).
Also XQEMU CI is probably going to do per-platform-workflows, so at least some maintainers will have slightly more expertise with that soon.

So it might be of interest to change this in this PR to per-platform-workflows (either by creating a duplicate of this PR for per-platform, *or* by sending a PR to @GXTX - if he'd be willing to accept such changes).

Reply:
I disagree enough about splitting per-OS and am not going to change this and have created an issue to track this at #243.